test/framework/alltypes_helpers.go

```diff
@@ -178,7 +184,7 @@ func dumpObject(resource runtime.Object, logPath string) {
 	namespace := metaObj.GetNamespace()
 	name := metaObj.GetName()
 
-	resourceFilePath := filepath.Clean(path.Join(logPath, namespace, kind, name+".yaml"))
+	resourceFilePath := filepath.Clean(path.Join(logPath, kind, namespace, name+".yaml"))
```

Comment:
I'm just trying to parse the artifacts folder and this change makes it very had:

After this change:
https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/9393/pull-cluster-api-e2e-full-main/1702754124908466176/artifacts/clusters/bootstrap/resources/

Before this change:
https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/periodic-cluster-api-e2e-release-1-5/1704077878771060736/artifacts/clusters/bootstrap/resources/

Before it was easy to go into a folder for a namespace / test case and go through various resources. Now it's pretty hard because it's always necessary to go up two levels. I prefer by far the grouping by namespace and then kind over first kind and then namespace

Can we revert this change?

@killianmuldoon @chrischdi Opinions?

Reply:
Fine to revert - I think you're right that namespace first is maps better onto how we've actually used this.

Reply:
PR: https://github.com/kubernetes-sigs/cluster-api/pull/9462